### DISCUSSION TITLE
Welcome to RescueAVR Discussions!

### DISCUSSION THREAD

**Discussion by felias-fogg** [felias-fogg/RescueAVR#5] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by xtal2k** [comment]
Hello. I'm trying to change lock bit of my ATTINY2312.  Though I read the datasheet of chips, I cannot change the settings properly....
I think I make a wrong circuit, and bad connections between pins.
Could you add a circuit diagram of ATTINY2312 version , please....

**Reply by felias-fogg** [reply]
Hi xtal2k,

I assume you are talking about the ATtiny2313 (not 2312, which does not exist). Two things:

1. If you only want to reset the lock bits and the chip responds to ISP commands, you can use the CHIP ERASE command to do so (see, e.g., https://onlinedocs.microchip.com/pr/GUID-EEA42155-A873-46B4-9500-D88617041B0B-en-US-2/index.html?GUID-DA9954DA-96B2-46E2-8835-576161B2280B). You need an ISP programmer and avrdude (or a similar program) for this purpose.
2. Concerning wiring, you need to connect the chip to the UNO in the same way as described for the ATmega on the Arduino UNO side. For example, A4 goes to RDY, A3 to OEi, etc., and 9 to 2 goes to DATA0 to DATA7, etc.   The only difference is that you do not need to connect XA1 (UNO pin 11). The pin layout of the 2313 is attached: 
![20pin-onboard](https://github.com/user-attachments/assets/0325700a-5e1a-452d-9753-cd9faff91e3a)
This means that you need to make the following connections:


| ATtiny Pin | Function                  | Arduino UNO Pin Name |
| ---------- | ------------------------- | -------------------- |
| 1          | RESETinv (12V switchable) |  (triggered by 13)                    |
| 2          |                           |                      |
| 3          | RDY                       | A4                   |
| 4          |                           |                      |
| 5          | XTAL1                     | A0                   |
| 6          | OEinv                     | A3                   |
| 7          | WRinv                     | A2                   |
| 8          | BS1                       | A1                   |
| 9          | XA0                       | 10                   |
| 10         | GND                       | GND                  |
| 11         | BS2                       | A5                   |
| 12         | DATA0                     | 9                    |
| 13         | DATA1                     | 8                    |
| 14         | DATA2                     | 7                    |
| 15         | DATA3                     | 6                    |
| 16         | DATA4                     | 5                    |
| 17         | DATA5                     | 4                    |
| 18         | DATA6                     | 3                    |
| 19         | DATA7                     | 2                    |
| 20         | Vcc (5V)                  | 12                  |


Note: I edited table row 1 and 12

**Reply by xtal2k** [reply]
Thank you for your reply !
Oh , made a mistake, I wrote a wrong chip name.
I will try to connect.
I would like to thank you for your kindness !

**Comment by felias-fogg** [comment]
Hi,

after going over my codes and docs again, I decided to design a PCB to make life easier. You can buy that now at Tindie.

**Reply by xtal2k** [reply]
Hi, 
Thank you for your announcement of new PCB.
I bought your PCB at Tindie , today.
 I would like to thank you for your kindness.

**Comment by Eltguy** [comment]
Hallo and thank you for that great stuff here. It has inspirered me for a new design even for TQFP32-components and a few improvements. If you are interested please look [here](https://github.com/Eltguy/SaveAVR).
regard Peter

**Reply by felias-fogg** [reply]
Hi Eltguy,

looks great! The only problem I see is that it is not trivial to remove a SMD MCU from the PCB. Do you use any reflow tool for that?

**Comment by Eltguy** [comment]
... desoldering is easier as you think, Bernhard. I use an air gun like [this](https://de.aliexpress.com/item/1005006773116240.html?spm=a2g0o.productlist.main.17.104fIlEgIlEgFo&algo_pvid=694d100c-eb7d-4407-816d-eaa9e8cda757&algo_exp_id=694d100c-eb7d-4407-816d-eaa9e8cda757-16&pdp_ext_f=%7B"order"%3A"673"%2C"eval"%3A"1"%7D&pdp_npi=4%40dis%21EUR%2151.57%2120.99%21%21%21416.22%21169.41%21%40211b430817504058090738818ed133%2112000038280173107%21sea%21DE%214239715761%21X&curPageLogUid=zk7DlQKQSKTQ&utparam-url=scene%3Asearch%7Cquery_from%3A). Heat up the housing from a distance of approx. 10 mm with a small nozzle and remove the chip with angled tweezers as soon as the tin has melted. It takes around 1min and makes no problems. You can also heat the chip from the backsite of the PCB, if possible. This protects the surrounding components.
I hope I don't need this anymore. I habve placed an additional ISP-Connector on the adapter board. Therefore I can set the fuses via ISP as usual (and repair it again with HV-programming if needed)  **before** I solder the chip on to the board. For myself is it an optimal solution.

**Comment by Eltguy** [comment]
Hallo Bernhard,
I have presented you my  saveAVR project with the TQFF32-adapter last month (look above). I'm testing now and I have a little issues with the terminal program V3.1.1. . The main features are working fine - start checkup, resurrect, chip erase, burn defauls and even restart. So the hardware is OK and the main goal is achieved. How ever the direkt changing of single lock-bits and fuses can't be programmed via HV? What do I wrong? Is there somthing I don't attend? Do you have any idea?
best regards
Peter

**Reply by felias-fogg** [reply]
You can only program the entire lock byte or fuse byte. Therefore, if you want to change a single bit, you must examine the current byte and then determine how changing that bit affects the byte. This can then be entered as a byte (in hex notation) with the appropriate command.  

For instance, if you want to set on an ATtiny85 bit 0 in the lock byte (LB1) to 0 and the lock byte is originally 0xFF, then you press `K` to change the lock byte and enter `FE`. This will program the LB1 bit.

Hope that helps

**Reply by Eltguy** [reply]
thank you for your quick response, Bernard. I was able to solve the problem now – a silly mistake! In HTERM, you have to appand explicitly a CR string at the end the hex input string. Just pressing the Enter key is not enough! The terminals of ARDUINO and Visual Studio Code do this automatically for you.
Thanks again
Peter

**Comment by Eltguy** [comment]
thank you for your quick response, Bernard. I was able to solve the problem now – a silly mistake! In HTERM, you have to appand explicitly a CR string at the end the hex input string. Just pressing the Enter key is unfortunately not enough! The terminals of ARDUINO and Visual Studio Code do this automatically for you.
Thanks again
Peter
